### PR TITLE
docs: remix section in tools

### LINKED ISSUE #866: Document Sapphire and Remix
See what is needed for Remix to work seamlessly with Sapphire. Currently, it only works via "universal" metamask connector. A section on this was written [years ago for Emerald](https://docs.oasis.io/dapp/emerald/writing-dapps-on-emerald#create-dapp-on-emerald-with-remix---ethereum-ide). Revamp it for Sapphire and move it out of Emerald into a separate chapter under https://docs.oasis.io/dapp/tools/ ➝ Remix.

OFC you won't be able to debug in Remix much against Sapphire because it won't reveal confidential information.

Don't forget to mention the transactions and queries are unencrypted and unsigned.

Don't forget to pinpoint solc version up to 0.8.24 (see https://github.com/rust-ethereum/evm/issues/277).

Also consider, if the Explorer➝Sourcify➝Remix flow is feasible to interact with a smart contract and worth describing it in this chapter how a (non-tech) user could communicate with the smart contract, e.g. if the maintainer decides to shut down the dApp frontend and the user is stuck with the contract and wants to withdraw his assets back.  FYI @lukaw3d built a simpler alternative https://abi-playground.oasis.io/ for this.

### PR DESCRIPTION
**Pull request by rube-de**
[PREVIEW](https://deploy-preview-1003--oasisprotocol-docs.netlify.app/dapp/tools/remix)

- new remix chapter in tools
- update screens
- add sections 
- ~add sourcify section~ (removed for now until sourcify fixes "view in remix" bug with their new plugin)
- add info about compiler / evm versions
- remove the remix section from emerald

### CODE REVIEW

File: docs/dapp/tools/remix.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+[#dev-central Discord channel][discord].
+
+[localnet]: ./localnet.mdx
+[network page]: https://docs.oasis.io/dapp/sapphire/network#rpc-endpoints
+[Remix]: https://remix.ethereum.org
+[Remix documentation]: https://remix-ide.readthedocs.io/en/latest/
+[metamask]: ../../general/manage-tokens/README.mdx#metamask
+[discord]: https://oasis.io/discord
+[ABI-Playground]: https://abi-playground.oasis.io/
+[issue]: https://github.com/oasisprotocol/docs/issues/866#issuecomment-2387513649
```
Comment (aefhm) [inline docs/dapp/tools/remix.md]: Hmm. This is one specific issue though?

File: docs/dapp/tools/remix.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+
+## Additional Tools
+
+### Sourcify Integration
+
+Verified contracts on Sourcify can be opened directly in Remix, enabling
+contract interaction without a dedicated frontend.
+
+:::info ABI-Playground
```
Comment (aefhm) [inline docs/dapp/tools/remix.md]:
```suggestion
:::info ABI Playground
```
(optional nit)

File: docs/dapp/tools/remix.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+   - EVM version: **`paris`** (found under Advanced Configuration)
+
+3. Click `Compile 1_Storage.sol`
+
+:::info Compiler Version
+
+The Sapphire uses the [Rust Ethereum EVM][rust-evm]. This implementation is
+compatible with Solidity versions up to **0.8.24**. However, it does not yet
+support some transaction types introduced in Solidity **0.8.25**, such as those
+mentioned in [rust-ethereum/evm#277][revm-277], which are pending release.
```
Comment (aefhm) [inline docs/dapp/tools/remix.md]:
```suggestion
mentioned in [rust-ethereum/evm#277][revm-277], which are underway
```

or maybe

```suggestion
mentioned in [rust-ethereum/evm#277][revm-277], pending release of the next version.
```

File: docs/dapp/tools/remix.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+:::
+
+## Prerequisites
+
+1. Install the [MetaMask browser extension][metamask]
+2. Configure your networks:
+   - Add Sapphire networks to MetaMask using the `Add to MetaMask` buttons on
+  our [network page]
+   - (Optional) Configure local network settings if you're using the Sapphire
```
Comment (aefhm) [inline docs/dapp/tools/remix.md:27]: (not relevant) to this PR, but maybe that could be a convenient local dev button to have

File: docs/dapp/tools/remix.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+account in the MetaMask and the corresponding Sapphire Network.
+
+## Working with Confidential Features
+
+Note that Remix operates without a Sapphire client, meaning transactions and
+queries are unencrypted and unsigned by default. To make use of Sapphire's
+confidential features, refer to our [Quickstart Tutorial].
+
+[Quickstart Tutorial]: https://docs.oasis.io/dapp/sapphire/quickstart
+
```
Comment (aefhm) [inline docs/dapp/tools/remix.md:91]:
```suggestion
```

**Review by aefhm — APPROVED** [review]

File: docs/dapp/tools/remix.md (hunk 1 above)
Comment (rube-de) [reply docs/dapp/tools/remix.md]: actually it's broken now totally? I can't open verified source code in Remix, even USDT on Ethereum just shows me then the default workspace in Remix. Does it work for you?

Comment (rube-de) [reply docs/dapp/tools/remix.md]: removed the section for now, as there is an issue with sourcify to open contracts in remix

File: docs/dapp/emerald/writing-dapps-on-emerald.mdx [hunk 6]
```diff
@@ -5,12 +5,18 @@ import {findSidebarItem} from '@site/src/sidebarUtils';
...
-a simpler web-only interface - the Remix IDE.
-Oasis Emerald exposes an EVM-compatible interface so writing dApps isn't much
+Oasis Emerald exposes an **EVM-compatible** interface so writing dApps isn't much
 different compared to the original Ethereum Network!
 
+We will walk you through the Hardhat configuration.
+
+Those who prefer a simpler web-only interface can also the Remix IDE, check our
+general [Remix guide] for that. Just remember to use the Emerald [networks]
+when selecting *Inject Web3* environment and connecting with your MetaMask.
```
Comment (aefhm) [inline docs/dapp/emerald/writing-dapps-on-emerald.mdx]:
```suggestion
Those who prefer a simpler web-only interface can also use the Remix IDE. Check out our
general [Remix guide]. Just remember to use the Emerald [networks]
when selecting *Inject Web3* environment and connecting to MetaMask.
```

File: docs/dapp/opl/celer/ping-example.md [hunk 7]
```diff
@@ -123,11 +123,11 @@ For compatibility with Sapphire, compile the contract using compiler version
...
-```solidity
-import "sgn-v2-contracts/contracts/message/framework/MessageBusAddress.sol";
-import "sgn-v2-contracts/contracts/message/framework/MessageReceiverApp.sol";
-import "sgn-v2-contracts/contracts/message/interfaces/IMessageBus.sol";
-```
+    ```solidity
+    import "sgn-v2-contracts/contracts/message/framework/MessageBusAddress.sol";
+    import "sgn-v2-contracts/contracts/message/framework/MessageReceiverApp.sol";
+    import "sgn-v2-contracts/contracts/message/interfaces/IMessageBus.sol";
+    ```
```
Comment (aefhm) [inline docs/dapp/opl/celer/ping-example.md:130]: Did you mean this identation?

```suggestion
```solidity
    import "sgn-v2-contracts/contracts/message/framework/MessageBusAddress.sol";
    import "sgn-v2-contracts/contracts/message/framework/MessageReceiverApp.sol";
    import "sgn-v2-contracts/contracts/message/interfaces/IMessageBus.sol";
```
```

File: docs/dapp/tools/remix.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+For comprehensive details about Remix's features, consult the
+[Remix documentation].
+
+:::
+
+## Prerequisites
+
+1. Install the [MetaMask browser extension][metamask]
+2. Configure your networks:
+   - Add Sapphire networks to MetaMask using the `Add to MetaMask` buttons on
```
Comment (aefhm) [inline docs/dapp/tools/remix.md]: (super nitpicky/not blocking) I wonder if a link back to the testnet faucet or mentioning that is the network used in the screenshot somehow could be helpful if someone came to this page first.

**Review by aefhm — APPROVED** [review]

File: docs/dapp/opl/celer/ping-example.md (hunk 7 above)
Comment (rube-de) [reply docs/dapp/opl/celer/ping-example.md:130]: the md linter expects it indented not fenced 🤷‍♂️

**Pull request merged**